pkg/server/app/commands/submit_transaction_test.go

```diff
@@ -1,45 +1,253 @@
 package commands_test
 
 import (
+	"bytes"
 	"context"
+	"encoding/json"
+	"errors"
+	"io"
 	"net/http"
 	"net/http/httptest"
+	"strings"
 	"testing"
 
 	"github.com/4chain-ag/go-overlay-services/pkg/core/engine"
 	"github.com/4chain-ag/go-overlay-services/pkg/server/app/commands"
 	"github.com/4chain-ag/go-overlay-services/pkg/server/app/jsonutil"
 	"github.com/bsv-blockchain/go-sdk/overlay"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
-// This is an example describing how the handler unit tests
-// should be structured and tested based on the HTTP standard package.
-func TestSubmitTransactionHandler_Handle(t *testing.T) {
-	// given:
-	app, err := commands.NewSubmitTransactionCommandHandler(submitTransactionProviderAlwaysOK{})
-	assert.NoError(t, err)
-	ts := httptest.NewServer(http.HandlerFunc(app.Handle))
+// SubmitTransactionProviderAlwaysSuccess is an implementation that always succeeds
+type SubmitTransactionProviderAlwaysSuccess struct{}
+
+func (SubmitTransactionProviderAlwaysSuccess) Submit(ctx context.Context, taggedBEEF overlay.TaggedBEEF, mode engine.SumbitMode, onSteakReady engine.OnSteakReady) (overlay.Steak, error) {
+	// Call the onSteakReady callback to simulate async completion
+	if onSteakReady != nil {
+		steak := overlay.Steak{}
+		onSteakReady(&steak)
+	}
+	return overlay.Steak{}, nil
+}
+
+// SubmitTransactionProviderAlwaysFailure is an implementation that always returns an error
+type SubmitTransactionProviderAlwaysFailure struct{}
+
+func (SubmitTransactionProviderAlwaysFailure) Submit(ctx context.Context, taggedBEEF overlay.TaggedBEEF, mode engine.SumbitMode, onSteakReady engine.OnSteakReady) (overlay.Steak, error) {
+	return overlay.Steak{}, errors.New("Submit transaction test error")
+}
+
+// SubmitTransactionProviderNeverCallback is an implementation that never calls the callback
+type SubmitTransactionProviderNeverCallback struct{}
+
+func (SubmitTransactionProviderNeverCallback) Submit(ctx context.Context, taggedBEEF overlay.TaggedBEEF, mode engine.SumbitMode, onSteakReady engine.OnSteakReady) (overlay.Steak, error) {
+	// Never call the callback which then should trigger the timeout
+	return overlay.Steak{}, nil
+}
+
+// For testing purposes only - allows creating a handler with a custom body limit using middleware type approach
+func createTestHandlerWithLimit(provider commands.SubmitTransactionProvider, limit int64) http.HandlerFunc {
+	handler, err := commands.NewSubmitTransactionCommandHandler(provider)
+	if err != nil {
+		panic(err)
```

Comment:
```suggestion
		panic(err)
```
I think now after refactoring commans and queries ... we avoiding of `panic` method .. maybe change function signature and return `(http.HandlerFunc, error)`